src/clients/repositories.rb

```diff
@@ -641,10 +642,20 @@ def buildList
             @displayed_service == Ops.get_string(srv_state, "alias", "")
         )
         ret = Builtins.add(ret, t)
-      end 
+      end
 
+      # there is some service, so allow to filter repos without service (bnc#944504)
+      if ret.size > 2
```

Comment:
`size > 2` is very non-obvious. how about `unless Builtins.isempty(@serviceStatesOut)`